docs/userguide/editing.md

```diff
@@ -27,73 +27,153 @@ rather than Google Sites.
 integer is fixed within any given platform.*
 
-## Bookmarks & Links
-
-For internal links, please select the Sites page as a destination rather than
-specifying the full URL as a generic web link.
-Unfortunately, Sites makes it hard to define an anchor on a page and create a
-link to that specific anchor (whether from the same page or another one).
+Unlike Github-flavored Markdown, in multi-line fenced code blocks, the programming
+language in the [info string](https://spec.commonmark.org/0.29/#info-string) 
+should be lowercased.
 
-*   For links to a specific section on the same page, please remove the link,
-    underline the former link text, and put "(§)" right after it.
-*   For links to a specific section on another page, just link to the page and
-    name the section. Please also put "(§)" right after it.
+## Bookmarks & Links
 
-If and when Sites offers a reasonable way of defining anchors and linking to
-them, we can search our pages for "(§)" and fix the links.
+For internal links, please used relative paths to the corresponding Markdown 
+file as the Markdown link path rather than specifying the full URL.
+
+Anchors are generated by Jekyll for sections using the section header text in
+lowercase with hyphens instead of spaces.
+
+> :construction: **TODO**: search our pages for "(§)" and fix the links.
+
+## Previewing Changes Locally
+
+Because of the latency in GitHub Pages rendering changes, it can be useful to 
+run a Jekyll instance locally to preview changes.
+
+### Setup
+
+GitHub Pages provides a list of
+[dependency versions](https://pages.github.com/versions/) used in order
+to allow local environments to match exactly. Since Jekyll is a Ruby project,
+setup details depend on gems (code packages), Bundler (dependency management for
+execution), and the version of Ruby.
+
+[rbenv](https://www.ruby-lang.org/en/documentation/installation/#rbenv) is one
+of the few different ways to install and control the version of Ruby.  To 
+install it and the required `ruby-build` plugin:
+```bash
+git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+cd ~/.rbenv && src/configure && make -C src # Optional but greatly speeds up certain operations
+echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
+```
+```bash
+mkdir -p "$(rbenv root)"/plugins
+git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+```
+
+To install specific Ruby version that GitHub Pages uses:
+```bash
+rbenv versions  # list all Ruby versions available
+rbenv install <version-num>  # install GH Pages Ruby version
+rbenv versions  # list all Ruby versions after installing
+```
+
+To activate the version of Ruby:
+```bash
+rbenv init  # OR:  eval "$(rbenv init -)"
+rbenv shell <version-num>
+```
+
+To install [Bunder](https://bundler.io/):
+```bash
+gem install bundler
+```
+
+The simplest way to ensure that
+transitive dependency versions are included correctly to just specify the
```

Comment:
This is not quite a sentence. Maybe insert "is" just before "to"?

Reply:
Done.